### PR TITLE
Unit tests for options.js

### PR DESCRIPTION
**Pull request by Hadegda**
## Description

Unit tests for options.js
And some small fixes in them

## Type of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] I have read [CONTRIBUTING](https://github.com/epam/miew/blob/master/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/epam/miew/blob/master/CODE_OF_CONDUCT.md) guides.
- [x] I have followed the code style of this project.
- [x] I have run `yarn run ci`: lint and tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation / The changes do not need docs update.

### CODE REVIEW

File: src/chem/selectors.test.js [hunk 1]
```diff
@@ -0,0 +1,1066 @@
...
+import dirtyChai from 'dirty-chai';
+import proxyquire from 'proxyquire';
+import selectors from './selectors';
+import Atom from './Atom';
+import ResidueType from './ResidueType';
+import utils from '../utils';
+
+chai.use(dirtyChai);
+
+describe('selectors', () => {
```
Comment (paulsmirnov) [inline src/chem/selectors.test.js]: The file is overwhelmingly big and hard to understand. So many helper variables. So many test cases. I think we can merge some tests together. I'm afraid I cannot provide useful comments right now, I'll look at the file and see what I can suggest/do.

File: src/options.js [hunk 2]
```diff
@@ -282,7 +280,10 @@ function _processArgsForURL(args) {
   if (!_.isArray(args)) {
     return args;
   }
-  return args[0] + (args.length < 2 ? '' : cOptsSep + _processOptsForURL(args[1]));
+  if (args.length < 2) {
+    return args[0];
+  }
+  return args[0] + cOptsSep + _processOptsForURL(args[1]);
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/options.js]: Thank you for your comment. I didn't notice this before

File: src/options.test.js [hunk 3]
```diff
@@ -339,5 +530,26 @@ describe('options', () => {
...
+
+  // TODO think about making public fromArray function and tests firstly it (not fromAttr or fromURL)
+  describe('.fromAttr(attr)', () => {
+    function urlize(opts) {
+      return `?${opts || ''}`;
+    }
+
+    it('restores high level mode properties', () => {
+      const tmp = options.fromURL(urlize(simpleRepsStr));
+      const tmp1 = options.fromAttr(simpleRepsStr);
```
[the remark this answers is not in the document]
Comment (paulsmirnov) [reply src/options.test.js]: I mean `tmp1` 😄

File: src/options.test.js [hunk 4]
```diff
@@ -142,6 +190,41 @@ describe('options', () => {
...
     + 'color:4291624959,dashSize:0.5&colorers.EL.carbon=-3&colorers.UN.color=16711680&'
     + 'camFov=45&camNear=0&resolution=lowest&autoResolution=true';
+  const overallCommands = [
+    'preset default',
+    'rep 0 m=BS c=EL',
+    'rep 1 s="sequence 85, 175, 256 and name CB and sequence 1:286" m=TX c=EL mt=SF',
+    'rep 2 m=TR radius=0.5 c=EL carbon=0',
+    'rep 3 m=CS subset="chain A and sequence 29 or serial 196"',
+    ' line "A.38.CO1" "A.38.CO2"',
+    ' line "A.38.CO1" "A.38.CO2" color=4291624959 dashSize=0.5',
```
[the remark this answers is not in the document]
Comment (paulsmirnov) [reply src/options.test.js]: Let's give it a try

File: src/options.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/options.test.js]: Oh, I fix it)

File: src/options.test.js [hunk 5]
```diff
@@ -339,5 +530,26 @@ describe('options', () => {
...
+    it('restores proper URL for complex with using dup for duplicate similar reps', () => {
+      expect(options.fromURL(urlize(duplicatedRepsStr))).to.equalOptions(duplicatedRepsOpts);
+    });
+
+    it('restores proper URL for complex with using double mode for duplicatee similar reps', () => {
+      expect(options.fromURL(urlize(doubleModeRepsStr))).to.equalOptions(duplicatedRepsOpts);
+    });
+  });
+
+  // TODO think about making public fromArray function and tests firstly it (not fromAttr or fromURL)
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/options.test.js]: I removed TODO here and in options.js
Also, I created a correspondent task: https://jira.epam.com/jira/browse/EPMCMBIMV-1287

File: src/options.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/options.test.js]: Solve the problem with extra spaces

[the remark this answers is not in the document]
Comment (Hadegda) [reply src/options.test.js]: Create a task on changing colors to hexes. It locates here: https://jira.epam.com/jira/browse/EPMCMBIMV-1288

**Comment by NataliaDSmirnova** [comment]
Thank you, guys, for this long-lasting task. I approve that tests for the options module. Selectors.test.js continues to grow in another task.

**Pull request merged**